### PR TITLE
gpui: Avoid redundant SharedString allocations in ShapedLine::split_at

### PR DESCRIPTION
**Pull request by cppcoffee**
Build split text directly from Arc<str> slices instead of allocating intermediate Strings. Reuse the original SharedString for full-line splits, return a static empty string for empty sides, and add a regression test covering boundary splits.

Release Notes:

- N/A

### CODE REVIEW

File: crates/gpui/src/text_system/line.rs [hunk 1]
```diff
@@ -208,8 +220,8 @@ impl ShapedLine {
         }
 
         // Split text
-        let left_text = SharedString::new(self.text[..byte_index].to_string());
-        let right_text = SharedString::new(self.text[byte_index..].to_string());
+        let left_text = self.text_slice(0, byte_index);
+        let right_text = self.text_slice(byte_index, self.text.len());
```
Comment (Veykril) [inline crates/gpui/src/text_system/line.rs]:
```suggestion
        let left_text = if byte_index == self.text.len() {
            self.text.clone()
        } else {
            SharedString::new(&self.text[..byte_index])
        };
        let right_text = if byte_index == 0 {
            self.text.clone()
        } else {
            SharedString::new(&self.text[byte_index..])
        };

```
is all we really need to change here to make this slightly more efficient. No need for that function or the test really.

**Review by Veykril — APPROVED** [review]

**Comment by Veykril** [comment]
Thanks!
